styles/Canonical/005-Industry-product-names.yml

```diff
@@ -0,0 +1,18 @@
+extends: substitution
+message: "Use '%s' instead of '%s'"
+link: https://docs.ubuntu.com/styleguide/en/#other-commonly-referenced-products/projects
+ignorecase: false
```

Comment:
You may want to consider ``ignorecase: true`` instead.

To explain why, let's review what the rule does with ``nVidia: NVIDIA``. 

If the ignorecase is false:
- ``NVidia`` or ``nvidia``, both common mistakes, will be ignored since they don't exactly match

If the ignorecase is true:
- All versions of ``NVIDIA`` that aren't written precisely in that manner will be corrected to it.

You could even change the rule to ``NVIDIA: NVIDIA`` and the result with ``ignorecase: true`` would be the same. All incorrect capitalizations would be fixed but ``NVIDIA`` would not trigger an error.

Reply:
Yes, the intention was to target some specific common mistakes and give a more useful tailored message. I think we need to rethink how these are applied though and also restrict the scope

Reply:
I could agree with targeting specific mistakes with other, more generalized, rules. However, when it comes to product names, there usually is only one right answer regardless of where the name is used. Whether it's the first word in a title or the last word in a sentence, ``Kubernetes`` has to start with a capital ``K`` and continue with lower case ``ubernetes``.

I'd argue that this difference warrants a more universal correction. Provided, of course, that the name is on its own. The rule assures that, as it is.

Reply:
That's not actually the case though, as 'maas' can be valid in context (e.g. when referring to a targetted substrate in Juju) or the name of a charm .
So I guess what I will probably do is change the scope and the ignorecase rule, I think that will be useful and not give too many false-positives

Reply:
I'm sorry, but the counter examples of "mass" is an exception, not a valid counter argument, nor an example

In development several words/terms only have one way to write them. For them, `ignorecase:true` is the norm.

If you have words, you don't want to fix the case, then it's another vale rule (so in another YAML file) that needs to be created with ignorecase:false.

So you would have to split.

But for me, the number of words were you can have 2 possible cases, doesn't need a rule, except if you want to fix typos.

My 2 cents.

Reply:
It doesn't just apply to 'maas' though, it applies to half the things on this list and the other one, so :shrug:
I already said I would change the ignorecase, but I will also restrict the scope.